pyart/retrieve/qpe.py

```diff
@@ -1,18 +1,644 @@
-#! /usr/bin/env python
+"""
+pyart.retrieve.qpe
+=========================================
+
+Functions for rainfall rate estimation
+
+.. autosummary::
+    :toctree: generated/
+
+    get_lowest_elevation
+    rr_zpoly
+    rr_z
+    rr_kdp
+    rr_a
+    rr_zkdp
+    rr_za
+    rr_hydro
+
+"""
+
 
 import os
 import sys
 from time import time
+from warnings import warn
 
 import numpy as np
 import netCDF4
 import matplotlib as mpl
 import pylab as pl
 
-import pyart.util.met as met
-#import ballsy_masked as ballsy
+from ..util import met
+from ..config import get_metadata, get_field_name, get_fillvalue
+# import ballsy_masked as ballsy
 
 
 def get_lowest_elevation(mmcg_ncf, req_moments, corrections):
     pass
-    #for each of the moments in req_moments fetch the lowest valid return
+    # for each of the moments in req_moments fetch the lowest valid return
+
+
+def rr_zpoly(radar, refl_field=None, rr_field=None):
+    """
+    Estimates rainfall rate from reflectivity using a polynomial Z-R relation
+    developed at McGill University
+
+    Parameters
+    ----------
+    radar : Radar
+        Radar object
+
+    refl_field : str
+        name of the reflectivity field to use
+
+    rr_field : str
+        name of the rainfall rate field
+
+    Returns
+    -------
+    rain : dict
+        Field dictionary containing the rainfall rate.
+
+    """
+
+    # parse the field parameters
+    if refl_field is None:
+        refl_field = get_field_name('reflectivity')
+    if rr_field is None:
+        rr_field = get_field_name('radar_estimated_rain_rate')
+
+    if refl_field in radar.fields:
+        refl = radar.fields[refl_field]['data']
+    else:
+        raise KeyError('Field not available: ' + refl_field)
+
+    refl2 = refl*refl
+    refl3 = refl*refl2
+    refl4 = refl*refl3
+
+    rr_data = np.ma.power(
+        10., -2.3+0.17*refl-5.1e-3*refl2+9.8e-5*refl3-6e-7*refl4)
+
+    rain = get_metadata(rr_field)
+    rain['data'] = rr_data
+
+    return rain
+
+
+def rr_z(radar, alpha=0.0376, beta=0.6112, refl_field=None, rr_field=None):
+    """
+    Estimates rainfall rate from reflectivity using a power law
+
+    Parameters
+    ----------
+    radar : Radar
+        Radar object
+
+    alpha,beta : floats
+        factor (alpha) and exponent (beta) of the power law
+
+    refl_field : str
+        name of the reflectivity field to use
+
+    rr_field : str
+        name of the rainfall rate field
+
+    Returns
+    -------
+    rain : dict
+        Field dictionary containing the rainfall rate.
+
+    """
+
+    # parse the field parameters
+    if refl_field is None:
+        refl_field = get_field_name('reflectivity')
+    if rr_field is None:
+        rr_field = get_field_name('radar_estimated_rain_rate')
+
+    if refl_field in radar.fields:
+        refl = radar.fields[refl_field]['data']
+    else:
+        raise KeyError('Field not available: ' + refl_field)
```

Comment:
OK. I misunderstood what you meant. Actually there are just two functions that use reflectivity AND rain rate. The others use different combinations. We could think about a generic function _parse_rrate_and_field but franckly speaking I think it would do more bad than good in terms of clarity. I did not see there was a check_field_exists function. I will simply change the if else by it.